### PR TITLE
WIP: Add Sony Xperia Z5 / msm8994 support

### PR DESCRIPTION
**Pull request by konradybcio**
Yes, this humongous dtb is needed for the bootloader to init display.

Yes, I've shrunk it down to the smallest size possible (it was >16k lines before).

Yes, I'm scared of possible eMMC damage but I also want to play with the bootloader.

### CODE REVIEW

File: dts/msm8994-sony-xperia-sumire.dts [hunk 1]
```diff
@@ -0,0 +1,1502 @@
...
+			reg-names = "dsi_ctrl\0dsi_phy\0dsi_phy_regulator\0mmss_misc_phys";
+			gdsc-supply = <0x27>;
+			vdd-supply = <0x2d>;
+			vddio-supply = <0x2e>;
+			vdda-supply = <0x2f>;
+			vcca-supply = <0x30>;
+			qcom,mdss-fb-map = <0x31>;
+			qcom,mdss-mdp = <0x32>;
+			clocks = <0x1e 0x618336ac 0x1e 0x684ccb41 0x1e 0xea30b0e7 0x1e 0xcc07d687 0x1e 0xf5a03f64 0x1e 0x3487234a 0x1e 0x28cafbe6 0x33 0x64a23fa0 0x33 0xf261a1a6 0x33 0x63154efc 0x33 0x1708ae85 0x33 0x5e69f8ef 0x33 0xa6b20c5a 0x34 0xfcd15658>;
+			clock-names = "mdp_core_clk\0iface_clk\0core_mmss_clk\0bus_clk\0byte_clk\0pixel_clk\0core_clk\0mdss_byte_clk_mux\0mdss_pixel_clk_mux\0byte_clk_src\0pixel_clk_src\0shadow_byte_clk_src\0shadow_pixel_clk_src\0clk_mdss_dsi1_vco_clk_src";
```
Comment (DolphinChips) [inline dts/msm8994-sony-xperia-sumire.dts:1273]: Maybe `sed 's/\\0/", "` here? ;)

File: dts/msm8994-sony-xperia-sumire.dts [hunk 2]
```diff
@@ -0,0 +1,1502 @@
...
+				somc,mdss-dsi-master;
+			};
+		};
+
+		qcom,mdss_dsi@fd998000 {
+			compatible = "qcom,mdss-dsi-ctrl";
+			label = "MDSS DSI CTRL->0";
+			cell-index = <0x00>;
+			reg = <0xfd998000 0x260 0xfd998500 0x280 0xfd998780 0x30 0xfd828000 0x108>;
+			reg-names = "dsi_ctrl\0dsi_phy\0dsi_phy_regulator\0mmss_misc_phys";
```
Comment (DolphinChips) [inline dts/msm8994-sony-xperia-sumire.dts:1264]: ditto

File: dts/msm8994-sony-xperia-sumire.dts [hunk 3]
```diff
@@ -0,0 +1,1502 @@
...
+			qcom,regs-dump-mdp = <0x1000 0x1408 0x2000 0x2064 0x2200 0x2264 0x2400 0x2464 0x2600 0x2664 0x2800 0x2864 0x5000 0x5130 0x5200 0x5230 0x7000 0x7130 0x7200 0x7230 0x9000 0x9130 0x9200 0x9230 0xb000 0xb130 0xb200 0xb230 0x15000 0x15130 0x15200 0x15230 0x17000 0x17130 0x17200 0x17230 0x19000 0x19130 0x19200 0x19230 0x1b000 0x1b130 0x1b200 0x1b230 0x25000 0x2512c 0x27000 0x2712c 0x45000 0x4538c 0x46000 0x4638c 0x47000 0x4738c 0x48000 0x4838c 0x49000 0x4938c 0x4a000 0x4a38c 0x55000 0x5522c 0x57000 0x5722c 0x59000 0x5922c 0x5b000 0x5b22c 0x65000 0x652b4 0x65800 0x65ab4 0x66000 0x662b4 0x66800 0x66ab4 0x67000 0x672b4 0x6b800 0x6ba54 0x6c000 0x6c254 0x6c800 0x6ca54 0x71000 0x710d0 0x71800 0x718d0>;
+			qcom,regs-dump-names-mdp = "MDP\0CTL_0\0CTL_1\0CTL_2\0CTL_3\0CTL_4\0VIG0_SSPP\0VIG0\0VIG1_SSPP\0VIG1\0VIG2_SSPP\0VIG2\0VIG3_SSPP\0VIG3\0RGB0_SSPP\0RGB0\0RGB1_SSPP\0RGB1\0RGB2_SSPP\0RGB2\0RGB3_SSPP\0RGB3\0DMA0_SSPP\0DMA1_SSPP\0LAYER_0\0LAYER_1\0LAYER_2\0LAYER_3\0LAYER_4\0LAYER_5\0DSPP_0\0DSPP_1\0DSPP_2\0DSPP_3\0WB_0\0WB_1\0WB_2\0WB_3\0WB_4\0INTF_1\0INTF_2\0INTF_3\0PP_0\0PP_1";
```
Comment (DolphinChips) [inline dts/msm8994-sony-xperia-sumire.dts:89]: ditto

File: dts/msm8994-sony-xperia-sumire.dts [hunk 4]
```diff
@@ -0,0 +1,1502 @@
...
+			qcom,mdss-ad-off = <0x79000 0x79800 0x7a000>;
+			qcom,mdss-highest-bank-bit = <0x03>;
+			qcom,mdss-has-decimation;
+			qcom,mdss-has-rotator-downscale;
+			qcom,mdss-has-bwc;
+			qcom,mdss-wfd-mode = "intf";
+			qcom,mdss-has-source-split;
+			qcom,mdss-ctl-len = <0x200>;
+			clocks = <0x1e 0x684ccb41 0x1e 0xcc07d687 0x1e 0x6dc1f8f1 0x1e 0x618336ac 0x1e 0x42a022d3>;
+			clock-names = "iface_clk\0bus_clk\0core_clk_src\0core_clk\0vsync_clk";
```
Comment (DolphinChips) [inline dts/msm8994-sony-xperia-sumire.dts:86]: ditto

File: dts/msm8994-sony-xperia-sumire.dts [hunk 5]
```diff
@@ -0,0 +1,1502 @@
...
+	soc {
+		#address-cells = <0x01>;
+		#size-cells = <0x01>;
+		ranges = <0x00 0x00 0x00 0xffffffff>;
+		compatible = "simple-bus";
+
+		qcom,mdss_mdp@fd900000 {
+			compatible = "qcom,mdss_mdp";
+			reg = <0xfd900000 0x90000 0xfd9c8000 0x4d4>;
+			reg-names = "mdp_phys\0vbif_phys";
```
Comment (DolphinChips) [inline dts/msm8994-sony-xperia-sumire.dts:21]: ditto

File: dts/msm8994-sony-xperia-sumire.dts [hunk 6]
```diff
@@ -0,0 +1,1502 @@
...
+			qcom,pmi8994@3 {
+				spmi-slave-container;
+				reg = <0x03>;
+				#address-cells = <0x01>;
+				#size-cells = <0x01>;
+
+				qcom,leds@d800 {
+					compatible = "qcom,qpnp-wled";
+					reg = <0xd800 0x100 0xd900 0x100 0xdc00 0x100 0xde00 0x100>;
+					reg-names = "qpnp-wled-ctrl-base\0qpnp-wled-sink-base\0qpnp-wled-ibb-base\0qpnp-wled-lab-base";
```
Comment (DolphinChips) [inline dts/msm8994-sony-xperia-sumire.dts:1365]: ditto

**Comment by Minecrell** [comment]
Can you adjust for the following commits:

- https://github.com/msm8916-mainline/lk2nd/commit/6309c37559fdc32e89285c1bb74b824e0b4f49c8 (common `lk2nd.mk`)
- https://github.com/msm8916-mainline/lk2nd/commit/6309c37559fdc32e89285c1bb74b824e0b4f49c8 (makes building i2c_gpio optional)
- https://github.com/msm8916-mainline/lk2nd/commit/9763281b94bdf28cdd17bb99f91fd14a905b42d4 (avoid having to add `ifneq ($(DISPLAY_USE_CONTINUOUS_SPLASH),1)`)

File: platform/msm_shared/scm.c [hunk 7]
```diff
@@ -1312,6 +1312,7 @@ static int scm_call2_atomic(uint32_t svc, uint32_t cmd, uint32_t arg1, uint32_t
 	return ret;
 }
 
+#if PLATFORM_USE_SCM_DLOAD
```
Comment (Minecrell) [inline platform/msm_shared/scm.c:1315]: Is this really necessary?

File: project/msm8994.mk [hunk 8]
```diff
@@ -12,27 +12,37 @@ else
...
-DEFINES += ABOOT_IGNORE_BOOT_HEADER_ADDRS=1
 
+DEFINES += ABOOT_IGNORE_BOOT_HEADER_ADDRS=1
 DEFINES += ABOOT_FORCE_KERNEL_ADDR=0x00008000
 DEFINES += ABOOT_FORCE_RAMDISK_ADDR=0x02000000
 DEFINES += ABOOT_FORCE_TAGS_ADDR=0x01e00000
 DEFINES += ABOOT_FORCE_KERNEL64_ADDR=0x00080000
 
 DEFINES += ENABLE_XPU_VIOLATION=1
+DEFINES += FBCON_DISPLAY_MSG=1
```
Comment (Minecrell) [inline project/msm8994.mk:42]: Unneeded, lk2nd.mk does that

File: project/msm8994.mk [hunk 9]
```diff
@@ -12,27 +12,37 @@ else
 DEBUG := 1
 endif
 
-EMMC_BOOT := 1
+
+#BE CAREFUL! SONY KITAKAMI DEVICES MIGHT SUFFER
+#PERMANENT EMMC DAMAGE
+EMMC_BOOT := 0
```
Comment (Minecrell) [inline project/msm8994.mk:18]: Avoid editing `msm8994.mk`, put everything extra into `msm8994-secondary.mk`.

File: README.md [hunk 10]
```diff
@@ -43,6 +44,7 @@ page on the EFIDroid wiki for an exact mapping of LK targets to SoCs.
 - Samsung Galaxy Tab 4 10.1 (2015) - SM-T533
 - Samsung Galaxy Tab A 8.0 LTE (2015) - SM-T357W
 - Samsung Galaxy Tab A 9.7 WiFi (2015) - SM-T550
+- Sony Xperia Z5 - sumire
```
Comment (Minecrell) [inline README.md:47]: some warning here would be nice...

File: platform/msm_shared/scm.c (hunk 7 above)
Comment (konradybcio) [reply platform/msm_shared/scm.c:1315]: Won't compile otherwise..

Comment (Minecrell) [reply platform/msm_shared/scm.c:1315]: I wonder if https://github.com/msm8916-mainline/lk2nd/commit/5e72e49ea0e027d4d526581c97d3a9437326b36a fixes it

File: project/msm8994.mk [hunk 11]
```diff
@@ -22,18 +23,23 @@ ENABLE_SMD_SUPPORT := 1
...
 DEFINES += ABOOT_FORCE_KERNEL_ADDR=0x00008000
 DEFINES += ABOOT_FORCE_RAMDISK_ADDR=0x02000000
 DEFINES += ABOOT_FORCE_TAGS_ADDR=0x01e00000
 DEFINES += ABOOT_FORCE_KERNEL64_ADDR=0x00080000
 
 DEFINES += ENABLE_XPU_VIOLATION=1
 
+#Enable the feature of long press power on
+DEFINES += LONG_PRESS_POWER_ON=1
+
```
Comment (Minecrell) [inline project/msm8994.mk:42]: There are still some edits in here :<

File: platform/msm_shared/scm.c (hunk 7 above)
Comment (konradybcio) [reply platform/msm_shared/scm.c:1315]: It does, that's what I get for not syncing before developing...

**Pull request closed without merging**